modules/redis/src/main/scala/zio/redis/options/Strings.scala

```diff
@@ -93,36 +97,24 @@ trait Strings {
 
   sealed case class BitPosRange(start: Long, end: Option[Long])
 
-  case object KeepTtl {
-    private[redis] def asString: String = "KEEPTTL"
-  }
-
-  type KeepTtl = KeepTtl.type
+  sealed trait SetExpire
 
-  sealed trait Expire { self =>
-    private[redis] final def asString: String =
-      self match {
-        case Expire.SetExpireSeconds      => "EX"
-        case Expire.SetExpireMilliseconds => "PX"
-      }
+  object SetExpire {
+    case object KeepTtl                                        extends SetExpire
+    case class SetExpireMilliseconds(duration: Duration)       extends SetExpire
+    case class SetExpireSeconds(duration: Duration)            extends SetExpire
+    case class SetExpireUnixTimeMilliseconds(instant: Instant) extends SetExpire
+    case class SetExpireUnixTimeSeconds(instant: Instant)      extends SetExpire
```

Comment:
Two things:
- keep case classes final
- since units are dictated by redis, I would replace `Duration` and `Instant` with `Long` and ensure they're valid